src/utils/cognitoAuthService.js

```diff
@@ -0,0 +1,61 @@
+const request = require('superagent');
+const jwt = require('jsonwebtoken');
+const jwkToPem = require('jwk-to-pem');
+const env = require('var');
+
+let auth_pem_keys = {};
+
+function validateToken(token, callback, pem_keys) {
+    const decodedJwt = jwt.decode(token, {complete: true});
+    if (!decodedJwt) {
+        console.log('Not a valid JWT token');
+        callback(new Error('Not a valid JWT token'));
+    }
+    const kid = decodedJwt.header.kid;
+    const pem = pem_keys[kid];
+    if (!pem) {
+        console.log('Invalid token');
+        callback(new Error('Invalid token'));
+    }
+    jwt.verify(token, pem, function (err, payload) {
+        if (err) {
+            console.log('Invalid Token.');
+            callback(new Error('Invalid token'));
+        } else {
+            // console.log('Valid Token.');
+            callback(null, 'Valid token', payload);
+        }
+    });
+}
+
+// https://janitha000.medium.com/authentication-using-amazon-cognito-and-nodejs-c4485679eed8
+module.exports.validate = function (token, callback) {
+    // first see if we have the JWT keys in the cache
+    if (Object.keys(auth_pem_keys).length) {
+        validateToken(token, callback, auth_pem_keys);
+    } else {
+        // get keys from Auth server
+        let url = `https://cognito-idp.${env.AUTH_COGNITO_POOL_REGION}.amazonaws.com/${env.AUTH_COGNITO_USER_POOL_ID}/.well-known/jwks.json`;
+        console.info('Getting JWT keys from: ' + url);
+        request
```

Comment:
There are libraries that do this whole functionality of grabbing the circulating pem files.  See `jwks-rsa`

Reply:
good point.  switched to using jwks-rsa